File: StringCalculator/src/test/java/github/io/junny/OperatorTest.java
package github.io.junny;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static github.io.junny.OperandEnum.*;

public class OperatorTest {

    @ParameterizedTest
    @CsvSource(value = {
            "2:+:3:5",
            "7:-:4:3",
            "6:*:5:30",
            "15:/:3:5"
    },delimiter = ':')
    void 값_계산_검증(String left, String operator, String right, Integer result){
        Assertions.assertThat(find(operator).operation(left,right)).isEqualTo(result);
    }
}

File: ladder/src/test/kotlin/io/github/junny/operatorTest.kt
package io.github.junny

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class operatorTest {

    @ParameterizedTest
    @CsvSource(value = [
        "2:3:5",
        "5:2:7"
    ], delimiter = ':')
    fun plusTest(left : Long, right : Long, result : Long){
        assertEquals(result, Operators.PLUS.func(left, right))
    }

    @ParameterizedTest
    @CsvSource(value = [
        "8:3:5",
        "9:2:7"
    ], delimiter = ':')
    fun minusTest(left : Long, right : Long, result : Long){
        assertEquals(result, Operators.MINUS.func(left, right))
    }
    @ParameterizedTest
    @CsvSource(value = [
        "8:3:24",
        "9:2:18"
    ], delimiter = ':')
    fun multiplyTest(left : Long, right : Long, result : Long){
        assertEquals(result, Operators.MULTIPLY.func(left, right))
    }
    @ParameterizedTest
    @CsvSource(value = [
        "8:2:4",
        "9:3:3"
    ], delimiter = ':')
    fun divideTest(left : Long, right : Long, result : Long){
        assertEquals(result, Operators.DIVIDE.func(left, right))
    }

    @ParameterizedTest
    @CsvSource(value = [
        "*:MULTIPLY",
        "-:MINUS",
        "/:DIVIDE",
        "+:PLUS"
    ], delimiter = ':')
    fun findOperatorTest(operator: Char, operators: Operators){
        assertEquals( Operators.findOperator(operator), operators)
    }

}